File: src/utils/logger.ts

export function logRequest(msg: string) {
  console.log('[REQ]', msg);
}

export function logError(msg: string) {
  console.error('[ERR]', msg);
}

export function createLogEntry(level: string, msg: string) {
  return { level, msg, ts: Date.now() };
}

File: src/utils/logger.cpp
#include "logger.h"
#include <iostream>

void Logger::log(const std::string& msg) {
    std::cout << "[LOG] " << msg << std::endl;
}

void Logger::error(const std::string& msg) {
    std::cerr << "[ERR] " << msg << std::endl;
}

void Logger::flush() {
    std::cout.flush();
}
